spec/bolt/node/winrm_spec.rb

```diff
@@ -203,19 +203,50 @@ def stub_winrm_to_raise(klass, message)
   end
 
   it "can run a task remotely", winrm: true do
-    contents = 'Write-Output "$env:PT_message_one" ${env:PT_message two}'
+    contents = 'Write-Host "$env:PT_message_one ${env:PT_message two}"'
     arguments = { :message_one => 'task is running',
                   :"message two" => 'task has run' }
     with_tempfile_containing('task-test-winrm', contents) do |file|
       expect(winrm._run_task(file.path, 'environment', arguments).value)
-        .to eq("task is running\r\ntask has run\r\n")
+        .to eq("task is running task has run\r\n")
+    end
+  end
+
+  it "will collect on stdout using valid output mechanisms", winrm: true do
+    contents = <<-PS
+    # explicit Format-Table for PS5+ overrides implicit Format-Table which
+    # includes a 300ms delay waiting for more pipeline output
+    # https://github.com/PowerShell/PowerShell/issues/4594
+    Write-Output "message 1" | Format-Table
```

Comment:
If my theory is correct, then this `Write-Output` call should never end up appearing in output *after* the other writes.

If my theory is incorrect, this is a decent trap for future us.